app/controllers/api/v1/restaurants_controller.rb

```diff
@@ -0,0 +1,23 @@
+class Api::V1::RestaurantsController < ApplicationController
+
+  def index
+    render json: { restaurant: Restaurant.all.sample }
+  end
+
+  def create
+    body = JSON.parse(request.body.read)
```

Comment:
In an ideal world, you shouldn't have to `JSON.parse()` the request body in an API controller action this way. The Rails `params` magic should handle that for you, assuming that everything has been correctly formulated.